starlite/types/builtin_types.py

```diff
@@ -0,0 +1,27 @@
+# nopycln: file
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+
+    from typing import Type, Union
+
+    from pydantic_factories.protocols import DataclassProtocol
+    from typing_extensions import TypeAlias, TypedDict
+
+
+__all__ = [
+    "DataclassType",
+    "DataclassTypeOrInstance",
+    "NoneType",
+    "TypedDictType",
+]
+
+DataclassType: "TypeAlias" = "Type[DataclassProtocol]"
+
+DataclassTypeOrInstance: "TypeAlias" = "Union[DataclassType, DataclassProtocol]"
+
+NoneType = type(None)
+
+# mypy issue: https://github.com/python/mypy/issues/11030
+TypedDictType: "TypeAlias" = "Type[TypedDict]"  # type:ignore[valid-type]
```

Comment:
I don't think we really want to make these public types, so I haven't exported them into the top level `starlite.types`, nor added to docs.

The typing of `TypedDictType` is the same as what pydantic does.

Added the `NoneType` alias as there were quite a few `type(None)` calls throughout lib and tests.

The dataclass types allowed for better typing of `utils.models` and `DTO` related stuff.

Also, using `TypeAlias` here is probably the right way:tm: to structure modules of aliases like this without the runtime imports rather than the:

```python
if TYPE_CHECKING:
    from somewhere import something
else:
    something = Any
```
...pattern that we have elsewhere in `starlite.types`.

Reply:
Indeed. 

We can inject globals there, but it's messy